core/contracts/financial-templates/implementation/FeePayer.sol

```diff
@@ -110,9 +112,24 @@ abstract contract FeePayer is Testable {
         );
         lastPaymentTime = time;
 
+        totalPaid = regularFee.add(latePenalty);
+        if (totalPaid.isEqual(0)) {
+            return totalPaid;
+        }
+        // If the effective fees paid as a % of the pfc is > 100%, then reduce it to 100%. The amount that `totalPaid` is greater than `_pfc` is
+        // reduced from the greater component of (regularFees, latePenalty).
+        if (totalPaid.isGreaterThan(_pfc)) {
```

Comment:
If we need to reduce `totalPaid` then we need to do case work because there are two components of `totalPaid`: `regularFees` and `latePenalty`. I decided to reduce the larger component by the diff between `totalPaid` and `_pfc`.

Am curious if anyone can think of better approaches to this.

Reply:
Will that always work? What if the regularFee and the latePenalty are each twice the pfc?

Something that I *think* works, but is kinda messy (at least the way I implemented it) is reducing the latePenalty first, and then reducing the regularFee:
```
if (totalPaid.isGreaterThan(_pfc)) {
  FixedPoint.Unsigned memory deficit = totalPaid.sub(_pfc);
  FixedPoint.Unsigned memory latePenaltyReduction = FixedPoint.min(latePenalty, deficit);
  latePenalty = latePenalty.sub(latePenaltyReduction);
  deficit = deficit.sub(latePenaltyReduction);
  regularFee = regularFee.sub(FixedPoint.min(regularFee, deficit));
  totalPaid = _pfc
}
```
This effectively pays the store first, then pays the late penalty if there's anything left over. There may be other better solutions, though.

Reply:
>Will that always work? What if the regularFee and the latePenalty are each twice the pfc?

Yeah that'll break it how I made it. Let me think about your solution

Reply:
I've been thinking about this a lot and I think I'm going crazy. I think the second to last line in your solution needs to change. What if the `deficit > latePenalty && (deficit - latePenalty > regularFee)`:

```
latePenaltyReduction = MIN(latePenalty, deficit) = latePenalty
latePenalty -= latePenaltyReduction
latePenalty -= latePenalty
latePenalty = 0
deficit -= latePenaltyReduction
deficit -= latePenalty
```

At this point the `latePenalty` is reduced to 0 and the `deficit` (minus the `latePenalty`) is still greater than the `regularFee`. Currently the results in the following calculation to `regularFee`:

```
regularFeeReduction = MIN(regularFee, deficit) = regularFee
regularFee -= regularFee = 0
```

This is odd because we `totalPaid = _pfc` but no fees are paid. 

I think changing the second to last line to this fixes it:

```
regularFee = MIN(regularFee, deficit)
```

Reply:
Actually disregard this. I think you're right